user/contactmethod/store.go

```diff
@@ -466,3 +477,39 @@ func (db *DB) FindAll(ctx context.Context, userID string) ([]ContactMethod, erro
 
 	return scanAll(rows)
 }
+
+func (db *DB) FindLastStatus(ctx context.Context, cmID string) (*n.MessageStatus, error) {
+	err := validate.UUID("Contact Method ID", cmID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = permission.LimitCheckAny(ctx, permission.User)
```

Comment:
Somehow we're going to want to ensure the requester is the same user as the contact method we're looking up